android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -337,12 +337,21 @@ constructor(
     fun formatDate(date: Date, expectedFormat: String = "E, MMM dd yyyy"): String {
       return date.formatDate(expectedFormat)
     }
+
+    fun generateRandomIntLengthSix(): Int {
+      val uniqueId =
+        System.currentTimeMillis().toString().takeLast(6).toInt() +
+          (100000..maxInger).shuffled().random()
+      return if (uniqueId.toString().length > 6) uniqueId.toString().takeLast(6).toInt()
+      else uniqueId
```

Comment:
I agree w/@ellykits, we can simplify this. Does the below work? Would be nice to move the constants elsewhere, but it seems odd to have them in the RulesEngine, but guess we could move to the companion object for now

```suggestion
      return Random.ints(inclusiveSixDigitMinimum, exclusiveSixDigitMaximum)
```

Reply:
@pld this is noted. What are the possibilities of `Random.ints(inclusiveSixDigitMinimum, exclusiveSixDigitMaximum) ` generating same number in two different devices?
That is why I thought of including last digits of timestamp as part of the return value

Reply:
Yes that's definitely possible, @dubdabasoduba said that this iteration we're ok w/that and we'll have to be sure any users know, so important to note that in the docs

Reply:
Great. This is noted @pld